include/openmc/interpolate.h

```diff
@@ -0,0 +1,95 @@
+
+#ifndef OPENMC_INTERPOLATE_H
+#define OPENMC_INTERPOLATE_H
+
+#include <cmath>
+#include <vector>
+
+#include "openmc/search.h"
+
+namespace openmc {
+
+inline double interpolate_lin_lin(
+  double x0, double x1, double y0, double y1, double x)
+{
+  return y0 + (x - x0) / (x1 - x0) * (y1 - y0);
+}
+
+inline double interpolate_lin_log(
+  double x0, double x1, double y0, double y1, double x)
+{
+  return y0 + log(x / x0) / log(x1 / x0) * (y1 - y0);
+}
+
+inline double interpolate_log_lin(
+  double x0, double x1, double y0, double y1, double x)
+{
+  return y0 * exp((x - x0) / (x1 - x0) * log(y1 / y0));
+}
+
+inline double interpolate_log_log(
+  double x0, double x1, double y0, double y1, double x)
+{
+  double f = log(x / x0) / log(x1 / x0);
+  return y0 * exp(f * log(y1 / y0));
+}
+
+inline double interpolate_lagrangian(const std::vector<double>& xs,
+  const std::vector<double>& ys, int idx, double x, int order)
+{
+  std::vector<double> coeffs;
+
+  for (int i = 0; i < order + 1; i++) {
+    double numerator {1.0};
+    double denominator {1.0};
+    for (int j = 0; j < order + 1; j++) {
+      if (i == j)
+        continue;
+      numerator *= (x - xs[idx + j]);
+      denominator *= (xs[idx + i] - xs[idx + j]);
+    }
+    coeffs.push_back(numerator / denominator);
+  }
+
+  return std::inner_product(
+    coeffs.begin(), coeffs.end(), ys.begin() + idx, 0.0);
+}
+
+double interpolate(const std::vector<double>& xs, const std::vector<double>& ys,
+  double x, Interpolation i = Interpolation::lin_lin)
+{
+  int idx = lower_bound_index(xs.begin(), xs.end(), x);
+
+  if (idx == xs.size())
+    idx--;
+
+  switch (i) {
+  case Interpolation::lin_lin:
```

Comment:
What about `Interpolation::histogram`? It seems like it would be nice to support that as well

Reply:
Yep. Easy enough!